src/caffeine/caffeine.c

```diff
@@ -21,26 +25,25 @@ static gex_Rank_t rank, size;
   const int double_Complex_workaround =4100;
 #endif
 
-void caf_caffeinate(mspace* symmetric_heap, intptr_t* symmetric_heap_start, mspace* non_symmetric_heap, gex_TM_t* initial_team)
+void caf_caffeinate(mspace* symmetric_heap, intptr_t* symmetric_heap_start, mspace* non_symmetric_heap, gex_TM_t** initial_team)
```

Comment:
Why did you add a level of indirection in this declaration? As you just added to the README, you can not/must not modify a `bind(C)` function declaration in one language without updating the corresponding declaration in the other language, because no tool will help you notice the discrepancy.

However I believe this change is Just Plain Wrong.

Reply:
I mention why I made this change in my comment I just barely posted above. Since `initial_team` is an `intent(out)` arg of type `c_ptr` in the BIND(C) declaration, I would expect it to be a double pointer on C side. I understand that I may be getting turned around with the passing of pointers back and forth between Fortran and C.

Reply:
The `c_ptr` in the Fortran declaration is the `gex_TM_t` itself (which is a typedef to a C pointer). Thus the `c_ptr intent(out)` becomes `gex_TM_t *`

Reply:
Please also add a comment on the C definition of `caf_caffeinate` clarifying this detail to help the next person who misreads what's going on here.

Reply:
Signature and definition has been updated and a note has been added.